### PR TITLE
Feature/0.45 pio upgrade

### PR DESCRIPTION
**Pull request by arnabmitra**
## Description

closes: #607 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: cmd/provenanced/cmd/root.go [hunk 1]
```diff
@@ -339,3 +342,15 @@ func overwriteFlagDefaults(c *cobra.Command, defaults map[string]string) {
 		overwriteFlagDefaults(c, defaults)
 	}
 }
+
+// getIAVLCacheSize If app.toml has an entry iavl-cache-size = <value>, default is 781250
+func getIAVLCacheSize(options servertypes.AppOptions) int {
+	// wish there was a
```
Comment (fkneeland-figure) [inline cmd/provenanced/cmd/root.go:348]: Can you get rid of this?

**Review by channa-figure — APPROVED** [review]
LGTM

File: cmd/provenanced/cmd/root.go (hunk 1 above)
Comment (arnabmitra) [reply cmd/provenanced/cmd/root.go]: done

**Review by fkneeland-figure — DISMISSED** [review]

**Comment by fkneeland-figure** [comment]
Testing an upgrade from `v1.7.6` to this.

**Review by dwedul-figure — APPROVED** [review]
LGTM.

**Review by iramiller — DISMISSED** [review]
Reviewed the 0.44.5->0.45 diff and verified changes required against this PR.  Looks complete from what I saw.

File: app/upgrades.go [hunk 2]
```diff
@@ -41,8 +40,11 @@ var handlers = map[string]appUpgrade{
 				}
 				return false
 			})
-			versionMap := make(module.VersionMap)
-			versionMap[metadatatypes.ModuleName] = 2
+
+			// Note: retreiving current module versions from upgrade keeper
```
Comment (fkneeland-figure) [inline app/upgrades.go:44]: #spellingnit ***retrieving***

**Review by fkneeland-figure — DISMISSED** [review]

File: app/upgrades.go (hunk 2 above)
Comment (arnabmitra) [reply app/upgrades.go]: fixed

**Comment by arnabmitra** [comment]
Because of the typo in the word reported, the approvals went away :(

**Review by iramiller — APPROVED** [review]

**Review by fkneeland-figure — APPROVED** [review]

**Pull request merged**